tools/run_tests/run_tests_matrix.py

```diff
@@ -141,6 +141,15 @@ def _create_test_jobs(extra_args=[], inner_jobs=_DEFAULT_INNER_JOBS):
                              extra_args=extra_args,
                              inner_jobs=inner_jobs)
 
+  for compiler in ['python3.6', 'python3.4']:
```

Comment:
I would get rid of all of this. Instead, add the new compilers to https://github.com/grpc/grpc/blob/master/tools/run_tests/run_tests.py#L671 - this should fix the failures in Linux Basictests